cli/src/commands/touch.rs

```diff
@@ -45,6 +45,12 @@ pub(crate) struct TouchArgs {
     )]
     revisions_opt: Vec<RevisionArg>,
 
+    /// Generate a new change-id
+    ///
+    /// This generates a new change-id for the revision.
+    #[arg(long)]
+    new_id: bool,
```

Comment:
nit: `--new-change-id`, `--update-change-id`, or `--reset-change-id`?

I don't know which one is better, but it would be nice if we can reuse the same verb as `--update-author`.